### PR TITLE
API to get locale-specific language display names 

### PR DESCRIPTION
**Pull request by snktd**

### CODE REVIEW

File: experimental/displaynames/src/displaynames.rs [hunk 1]
```diff
@@ -83,3 +90,80 @@ impl DisplayNames {
...
+}
+
+impl LanguageDisplayNames {
+    /// Creates a new [`LanguageDisplayNames`] from locale data and an options bag.
+    ///
+    /// [📚 Help choosing a constructor](icu_provider::constructors)
+    /// <div class="stab unstable">
+    /// ⚠️ The bounds on this function may change over time, including in SemVer minor releases.
+    /// </div>
+    pub fn try_new_language_unstable<D: DataProvider<LanguageDisplayNamesV1Marker> + ?Sized>(
```
Comment (sffc) [inline experimental/displaynames/src/displaynames.rs:128]:
```suggestion
    pub fn try_new_unstable<D: DataProvider<LanguageDisplayNamesV1Marker> + ?Sized>(
```

File: experimental/displaynames/src/displaynames.rs [hunk 2]
```diff
@@ -68,12 +75,12 @@ impl DisplayNames {
...
-    pub fn of(&self, region_code: &str) -> Option<&str> {
-        match <TinyAsciiStr<3>>::from_str(region_code) {
+    pub fn of(&self, code: &str) -> Option<&str> {
+        match <TinyAsciiStr<3>>::from_str(code) {
             Ok(key) => {
-                let data = self.data.get();
+                let data = self.region_data.get();
                 match self.options.style {
                     Style::Short => data.short_names.get(&key),
                     _ => data.names.get(&key),
```
Comment (sffc) [inline experimental/displaynames/src/displaynames.rs:86]: Note: Both here and below, we should do the fallback from short to long automatically. We could optionally have another function, like `of_strict`, that does not fall back

File: experimental/displaynames/src/options.rs [hunk 3]
```diff
@@ -48,11 +67,12 @@ pub enum Style {
     Short,
     Long,
     Menu,
+    None,
```
Comment (sffc) [inline experimental/displaynames/src/options.rs:70]: Suggestion: `Option<Style>` and resolve the default in the individual formatter

File: experimental/displaynames/src/displaynames.rs (hunk 1 above)
Comment (snktd) [reply experimental/displaynames/src/displaynames.rs]: Done for both language and region.

File: experimental/displaynames/src/displaynames.rs (hunk 2 above)
Comment (snktd) [reply experimental/displaynames/src/displaynames.rs]: Added the change to fallback to the names without any alt variants. I can add a function of_strict later if needed.

File: experimental/displaynames/src/options.rs (hunk 3 above)
Comment (snktd) [reply experimental/displaynames/src/options.rs]: Makes sense. Added the change.

**Review by sffc — APPROVED** [review]

**Pull request merged**